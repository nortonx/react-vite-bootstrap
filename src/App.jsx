import './scss/styles.scss';
import { Routes, Route } from 'react-router-dom';
import MainLayout from './layouts/MainLayout';
import NotFound from './components/NotFound.jsx';
import Home from './routes/Home';
import Form from "./routes/Form";
import About from './routes/About';


function App() {
  return (
    <Routes>
      <Route path="/" element={<MainLayout />}>
        <Route index element={<Home />} />
        <Route path="form" element={<Form />} />
        <Route path="about" element={<About />} />
        <Route path="*" element={<NotFound />} />
      </Route>
    </Routes>
  );
}

export default App;
