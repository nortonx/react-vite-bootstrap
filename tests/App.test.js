import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import App from "../src/App";

describe("App", () => {
  it("Renders App component", async () => {
    render(<App />)
    const homeElement = await screen.findByTestId("home");
    expect(homeElement).toBeInTheDocument();
  });

  it("Should render and match snapshot", () => {
    const element = renderer.create(<App />);
    expect(element).toMatchSnapshot();
  })
})