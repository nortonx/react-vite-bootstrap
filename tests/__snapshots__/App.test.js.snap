// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders App component 1`] = `
<div
  class="container py-4 px-3 mx-auto"
  data-testid="main-layout"
>
  <div
    class="container header"
    data-testid="header"
  >
    <header
      class="d-flex flex-wrap justify-content-center py-3 mb-4 border-bottom"
    >
      <a
        class="d-flex align-items-center mb-3 mb-md-0 me-md-auto text-dark text-decoration-none"
        href="/"
      >
        Brand
      </a>
      <ul
        class="nav nav-pills"
      >
        <li
          class="nav-item"
        >
          <a
            activeclassname="active"
            aria-current="page"
            class="nav-link active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            activeclassname="active"
            class="nav-link"
            href="/about"
          >
            About
          </a>
        </li>
      </ul>
    </header>
  </div>
  <div
    class="container home"
    data-testid="home"
  >
    <h1>
      This is the Home content
    </h1>
  </div>
  <div
    class="container footer"
    data-testid="footer"
  >
    <footer
      class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top"
    >
      <div
        class="col-md-4 d-flex align-items-center"
      >
        <a
          class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1"
          href="/"
        >
          Logo
        </a>
        <span
          class="mb-3 mb-md-0 text-muted"
        >
          2023
           Â©
        </span>
      </div>
      <ul
        class="nav col-md-4 justify-content-end list-unstyled d-flex"
      >
        <li
          class="ms-3"
        >
          <a
            class="text-muted"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="ms-3"
        >
          <a
            class="text-muted"
            href="/about"
          >
            About
          </a>
        </li>
      </ul>
    </footer>
  </div>
</div>
`;
